crates/ruff_python_literal/src/format.rs

```diff
@@ -282,6 +284,41 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
     })
 }
 
+/// Parses a format part within a format spec
+fn parse_nested_format_part(text: &str) -> Result<(Option<FormatPart>, &str), FormatSpecError> {
+    let mut end_bracket_pos = None;
+    let mut left = String::new();
+
+    if text.is_empty() {
+        return Ok((None, text));
+    }
+
+    for (idx, c) in text.char_indices() {
+        if idx == 0 {
+            if c != '{' {
+                return Ok((None, text));
+            }
+        } else if c == '{' {
+            return Err(FormatSpecError::ReplacementNotAllowed);
+        } else if c == '}' {
+            end_bracket_pos = Some(idx);
+            break;
```

Comment:
Would it be possible to inline the `if let Some(pos) = end_bracket_pos)` case by using a return instead of assigning to a variable and using break?

Reply:
This is copied from the implementation at https://github.com/astral-sh/ruff/blob/8358c80e0e5decf850388b92629e5b04a4ee5f90/crates/ruff_python_literal/src/format.rs#L1068-L1104 — should we change them both?